examples/shapes/src/main.rs

```diff
@@ -0,0 +1,194 @@
+    event_loop
+        .run(move |event, event_loop| match event {
+            Event::WindowEvent {
+                ref event,
+                window_id,
+            } => {
+                let Some(render_state) = &mut render_state else {
+                    return;
+                };
+                if render_state.window.id() != window_id {
+                    return;
+                }
+                match event {
+                    WindowEvent::CloseRequested => event_loop.exit(),
+                    WindowEvent::Resized(size) => {
+                        // Resize the surface when the window is resized
+                        render_cx.resize_surface(&mut render_state.surface, size.width, size.height)
+                    }
+                    WindowEvent::RedrawRequested => {
+                        // This is where all the rendering happens
+
+                        // Get the window size
+                        let width = render_state.surface.config.width;
+                        let height = render_state.surface.config.height;
+                        // Get a handle to the device
+                        let device_handle = &render_cx.devices[render_state.surface.dev_id];
+                        // Use the render_state to retrieve the surface to draw to
+                        let surface_texture = render_state
+                            .surface
+                            .surface
+                            .get_current_texture()
+                            .expect("failed to get surface texture");
+
+                        // Define the render parameters.
+                        let render_params = vello::RenderParams {
+                            // Background color
+                            base_color: Color::BLACK,
+                            // Width
+                            width,
+                            // Height
+                            height,
+                            // Antialiasing method to use. Other methods: AaConfig::Area, AaConfig::Msaa8
+                            antialiasing_method: AaConfig::Msaa16,
+                        };
+
+                        // Create some shapes!
+                        let stroke = Stroke::new(6.0);
+
+                        let rect = RoundedRect::new(10.0, 10.0, 240.0, 240.0, 20.0);
+                        let rect_stroke_color = Color::rgb(0.9804, 0.702, 0.5294);
+
+                        let circle = Circle::new((420.0, 200.0), 120.0);
+                        let circle_fill_color = Color::rgb(0.9529, 0.5451, 0.6588);
+
+                        let ellipse = Ellipse::new((250.0, 420.0), (100.0, 160.0), -90.0);
+                        let ellipse_fill_color = Color::rgb(0.7961, 0.651, 0.9686);
+
+                        let line = Line::new((260.0, 20.0), (620.0, 100.0));
+                        let line_stroke_color = Color::rgb(0.5373, 0.7059, 0.9804);
+
+                        // Draw the shapes!
```

Comment:
Where does the scene get reset? My reading of this code suggests that the same shapes will be drawn on top each other, repeated for each frame.

This would mean performance would get worse as the program runs, which isn't really the impression we want to give

Reply:
look one line above

Reply:
Thanks, although that was added in response to my comment :laughing:

Reply:
My question would be why resetting and making new in the first place